src/app/core/user/user.ts

```diff
@@ -34,7 +34,23 @@ export class User extends Entity {
     label: $localize`:Label of username:Username`,
     validators: { required: true },
   })
-  name: string;
+  set name(value: string) {
+    if (this._name) {
+      // Throwing error if trying to change existing username
+      const label = User.schema.get("name").label;
+      throw new Error(
+        $localize`:Error message when trying to change the username|e.g. username cannot be changed after initialization:${label} cannot be changed after initialization`
```

Comment:
best practices would suggest to not mix logic (the Entity model here) with UI (translating an error message), I think. The clean but more complex approach would be to define error codes or classes and only put human-readable and translated error messages when displaying it to the user. But I have my doubts that complexity is a worthy trade off here. :+1:

Reply:
Definitely a good thought. At the moment we are missing a general *display error* approach where we could translate these errors to a user friendly name.